exchanges/graphcache/src/operations/write.ts

```diff
@@ -247,7 +247,7 @@ const writeSelection = (
     } else if (entityKey && !isRoot) {
       // This is a leaf node, so we're setting the field's value directly
       InMemoryData.writeRecord(entityKey || typename, fieldKey, fieldValue);
-    }
+    } else if (ctx.optimistic && isRoot) continue;
```

Comment:
Should we leave a TODO here for scalar optimistic results or just address that in a subsequent PR?
I’d have time to take a look at that, because I suppose it’s expected behaviour 🤷‍♀️

Reply:
I'd like to defer this to a later PR